packages/components/src/CopyButton.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copy button that has an icon change when clicked, and copies the string passed in.
+ *
+ * props:
+ * @param {string} copy the value to copy when clicked
+ * @param {string} ButtonKind the kind of button
+ * @param {string} tooltip optional tooltip label ex. 'Copy column name'
+ * @param {string} className optional extra classname
+ * @param {string} data-testid optional extra testid
+ *
+ *
+ */
+
+import React from 'react';
+import { vsPassFilled, vsCopy } from '@deephaven/icons';
+import { useCopyToClipboard } from '@deephaven/react-hooks';
+import Button, { ButtonKind } from './Button';
+
+type CopyButtonProps = {
+  copy: string;
+  kind?: ButtonKind;
+  tooltip?: string;
+  className?: string;
+  'data-testid'?: string;
+  children?: React.ReactNode;
```

Comment:
Should have an optional error handler for if the copy fails. It's possible for execCommand to fail if it's called from a timer, for example: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard#using_execcommand
There may be other cases with a permission locked browser, not sure.

Reply:
Copy button should handle it internally, not users of the component. And I don't think I have anything I want to do here other than the checkmark never appears, which is what it does.